mono/metadata/handle.c

```diff
@@ -319,14 +316,11 @@ mono_handle_stack_scan (HandleStack *stack, GcScanFunc func, gpointer gc_data, g
 		check_handle_stack_monotonic (stack);
 
 	/*
-	  We're called twice - on the imprecise pass we do nothing.
-	  Interior pointers are retained in managed frames.
-	  On the precise pass, we scan all the objects where the handles point to the start of
-	  the object.
-
-	  Note that if we're running, we know the world is stopped.
-	*/
-	if (!precise)
+	 * We're called twice, on the precise pass we do nothing.
+	 * On the inprecise pass, we pin the objects pointed to by the handles.
+	 * Note that if we're running, we know the world is stopped.
+	 */
+	if (precise)
```

Comment:
I think this is too subtle.
Can the comment say more like:
We are called twice, first with !precise, second with precise (or whatever is true).
Being called with precise was historically useful.
Now everything needed is on the imprecise pass, and nothing is done on the precise pass.
In the imprecise pass, we pin the objects pointed to by the handles.